### PR TITLE
fix(launch): Fill macros in resource arguments to SageMaker

### PR DESCRIPTION
**Pull request by KyleGoyette**
Description
-----------

- Fixes WB-NNNNN
- Fixes #NNNN

Fixes the `fill_macros` call in the SageMakerRunner to actually use the replaced values for the macros.

- [ ] I updated CHANGELOG.md, or it's not applicable

Testing
-------
Unit tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ Please add to the relevant subsections under Unreleased below on every PR where
 
 - Handle `path_prefix`es that don't correspond to directory names when downloading artifacts by @moredatarequired in https://github.com/wandb/wandb/pull/7721
 - Fix creating or updating an artifact with the `incremental=True` flag by @amusipatla-wandb in https://github.com/wandb/wandb/pull/7939
+- Fill macros in resource arguments to SageMaker by @KyleGoyette https://github.com/wandb/wandb/pull/7993
```
Comment (kptkin) [inline CHANGELOG.md:27]: maybe a little bit more context? for example for me it's hard to parse what this fix does

**Review by kptkin — APPROVED** [review]

**Review by TimH98 — APPROVED** [review]

**Review by kptkin — APPROVED** [review]
